scripts/check-data-freshness.ts

```diff
@@ -32,6 +33,42 @@ async function checkUrlDataAvailability(
   return content.status;
 }
 
+// Returns the response data obtained by trying to request the URL content.
+async function returnAvailablePdfUrlData(
```

Comment:
My gut tells me this at least some parts of this method should be consolidated with the non-PDF fetching above. Are there compelling reasons not to do that?